lib/utils/middlewares/context_middleware.ts

```diff
@@ -1,8 +1,20 @@
+import * as express from "express";
+
 import { right } from "../either";
 
 import { IRequestMiddleware } from "../request_middleware";
 
-import { IContext, IRequestWithContext } from "azure-function-express-cloudify";
+import { IContext } from "azure-function-express";
+
+const CONTEXT_IDENTIFIER = "context";
+
+export function setAppContext(app: express.Express, context: IContext<{}>): void {
+  app.set(CONTEXT_IDENTIFIER, context);
+}
+
+export function getAppContext<T>(request: express.Request): IContext<T> {
+  return request.app.get(CONTEXT_IDENTIFIER);
```

Comment:
the return type is actually `IContext<T> | undefined`

Reply:
according to the comment "TODO: validate that the context is indeed defined, respond with ResponseErrorInternal instead"

does it return ResponseErrorInternal  if we throw an Exception ?

because in this case we can just throw, leave the return type to IContext<T> and the check will be satisfied (this is a good spot for throwing)

Reply:
ok I've checked it and it returns internel server error in case context is not found:

wrapRequestHandler|ERROR|http://localhost:7071/api/v1/messages/... |Error: Cannot get context from request